iosApp/iosApp/Pages/Map/HomeMapViewHandlerExtension.swift

```diff
@@ -80,25 +69,54 @@ extension HomeMapView {
         lastNavEntry = navigationStack.last
     }
 
-    func loadGlobalData() {
-        Task(priority: .high) {
-            await activateGlobalListener()
+    @MainActor
+    func activateGlobalListener() async {
+        for await globalData in globalRepository.state {
+            self.globalData = globalData
         }
+    }
+
+    func fetchGlobalData() {
         Task {
             await fetchApi(
                 errorBannerRepository,
                 errorKey: "HomeMapView.loadGlobalData",
                 getData: { try await globalRepository.getGlobalData() },
-                onRefreshAfterError: loadGlobalData
+                onRefreshAfterError: fetchGlobalData
             )
         }
     }
 
+    func loadGlobalData() {
+        Task(priority: .high) {
+            await activateGlobalListener()
+        }
+        fetchGlobalData()
+    }
+
     @MainActor
-    func activateGlobalListener() async {
-        for await globalData in globalRepository.state {
-            self.globalData = globalData
+    func activateRouteShapeListener() async {
+        for await railRouteShapes in railRouteShapeRepository.state {
+            self.railRouteShapes = railRouteShapes
+        }
+    }
+
+    func fetchRouteShapes() {
+        Task {
+            await fetchApi(
+                errorBannerRepository,
+                errorKey: "HomeMapView.handleAppear",
+                getData: { try await railRouteShapeRepository.getRailRouteShapes() },
+                onRefreshAfterError: fetchRouteShapes
+            )
+        }
+    }
+
+    func loadRouteShapes() {
+        Task(priority: .high) {
```

Comment:
Do we set task priority anywhere else in the app?

Reply:
I started using it as part of the global data caching when performing map layer tasks, and that's the only other place we use it currently. There were a few times where it seemed like the tasks were getting delayed a weird amount, so I tried adding the priority explicitly. It's probably not strictly necessary, but we do want these to be executed as quickly as possible so that the map layers appear asap.